Source/Core/MainButton_Window/PrisonerButtonWindow.cs

```diff
@@ -82,71 +82,59 @@ private PrisonersTabDef CurTab
 
         public PrisonerButtonWindow() : base()
         {
-           foreach(var def in DefDatabase<PrisonersTabDef>.AllDefs)
+           foreach(var tabDef in DefDatabase<PrisonersTabDef>.AllDefs)
            {
-                DebugLogger.debug($"Def: {def.defName}, def.dev: {def.dev}, dev: {Prefs.DevMode}");
-                if(def.dev == false)
-                {
-                    tabsView.Add(def, CreateWindow(def));
-                }
-                if (def.dev == true && Prefs.DevMode)
-                {
-                    tabsView.Add(def, CreateWindow(def));
-                }
-
-            }
+                DebugLogger.debug($"Def: {tabDef.defName}, def.dev: {tabDef.dev}, dev: {Prefs.DevMode}");
+                tabsView[tabDef] = CreateWindow(tabDef);
+           }
            DebugLogger.debug("Prisoners main windows constructed");            
         }
 
         public override void DoWindowContents(Rect inRect)
         {           
-            
-            Text.Anchor = TextAnchor.UpperLeft;
-            Text.Font = GameFont.Small;   
-            
-            inRect.yMin += 32f;
+            inRect.yMin += TabDrawer.TabHeight;
             Widgets.DrawMenuSection(inRect);
             TabDrawer.DrawTabs(inRect, tabs);
-            inRect.yMin += TabDrawer.TabHeight;           
-            GetTable(CurTab).DoWindowContents(inRect);
+            inRect.yMin += Margin;           
+            if (Event.current.type != EventType.Layout)
+            {
+                GetTable(CurTab).DoWindowContents(inRect);
+            }
         }
         public override void PostOpen()
         {
             base.PostOpen();
             tabs.Clear();
-            foreach (PrisonersTabDef tabDef in DefDatabase<PrisonersTabDef>.AllDefs.Where(def => !def.dev || def.dev == Prefs.DevMode).OrderBy(def => def.order))
+            foreach (var tabDef in tabsView.Keys.Where(d => !d.dev || d.dev == Prefs.DevMode).OrderBy(d => d.order))
```

Comment:
logic from `where` is used in few places - please extract it to dedicated method